### DISCUSSION TITLE
[Bug]: Unable to load pimcore_enterprise_subscription_tools on Composer installation

### DISCUSSION THREAD

**Discussion by NiklasBr** [pimcore/pimcore#11897] [General]
### Expected behavior

If `composer.json` contains:
```json
    "repositories": {
        "pimcore_enterprise": {
            "type": "composer",
            "url": "https://enterprise.repo.pimcore.com/"
        },
    }
```
And `config/config.yaml` contains:
```yaml
pimcore_enterprise_subscription_tools:
    token: '%env(PIMCORE_ENTERPRISE_TOKEN)%'
    production_system: '%env(PIMCORE_ENTERPRISE_IS_PROD)%'
```
And the token has been added to composer global auth, and env file, then `composer update pimcore/pimcore` should work

### Actual behavior

```
Script Pimcore\Composer::clearCache handling the pimcore-scripts event terminated with an exception


  [RuntimeException]
  An error occurred when executing the "/usr/local/bin/php bin/console --ansi cache:clear --no-warmup" command:

  Exit code: 255

  Symfony\Component\Config\Exception\LoaderLoadException {#1423
    #message: "There is no extension able to load the configuration for "pimcore_enterprise_subscription_tools" (in "/var/www/pimcore/con
  fig/packages/dev/../../config.yaml"). Looked for namespace "pimcore_enterprise_subscription_tools", found ""framework", "security", "tw
  ig", "monolog", "doctrine", "doctrine_migrations", "sensio_framework_extra", "cmf_routing", "presta_sitemap", "scheb_two_factor", "fos_
  js_routing", "flysystem", "debug", "web_profiler", "pimcore", "pimcore_admin", "workflow", "web2_print_tools", "output_data_config_tool
  kit", "pimcore_ecommerce_framework", "frontend_permission_toolkit", "elements_process_manager", "pimcore_data_hub", "pimcore_data_impor
  ter", "snc_redis", "ch_cookie_consent", "hotjar", "algolia"" in /var/www/pimcore/config/packages/dev/../../config.yaml (which is being
  imported from "/var/www/pimcore/config/packages/dev/config.yaml")."
```

### Steps to reproduce

Follow the instructions sent when purchasing Enterprise Subscription.

**Comment by fashxp** [comment]
instructions say: 

- Add token authentication to composer
- Add repository to composer.json
- Install enterprise extensions as described in their documentation (composer and additional installation steps as described in the extension documentation)
- **After** installation, add the token to a configuration file ... 

and this will work...

**Reply by NiklasBr** [reply]
Are you sure? Did we get the wrong instructions? The step for this says "Prepare your Installation" and the section after the preparation says we should run the composer update.

**Reply by fashxp** [reply]
screenshot of the license pdf

![image](https://user-images.githubusercontent.com/8792145/162968520-98357f2a-63a4-412d-beba-1cd0f3160d0c.png)

**Reply by NiklasBr** [reply]
Looks like ours, yes. So, under the *"Prepare your Installation"* we prepare it, and then the step *below* that one, there is an instruction to run `composer update pimcore/pimcore` which is where the error happens.

**Reply by fashxp** [reply]
But then you didn't install any enterprise extensions. 

What you are talking about is a different section. It is about using Pimcore LTS. If you are just using LTS (without any extensions), no token config is needed (for now). 

![image](https://user-images.githubusercontent.com/8792145/162969905-7f2267f9-d02e-45af-96ce-19a8fe6ccabf.png)

**Reply by NiklasBr** [reply]
If I understand you correctly, these steps are only for LTS, but then if we want to install the Pimcore Enterprise Extension, who do we speak to to receive that documentation?

**Reply by fashxp** [reply]
no, you don't understand me correctly - or trying to misunderstand me. 

to install enterprise extension follow instructions of first screenshot, to use pimcore lts follow instructions of second screenshot. not too complicated.

**Reply by NiklasBr** [reply]
You mean this part?
`# additional installation steps as described in the extension documentation`

I thought that was a placeholder like "additional steps to be written here" sort of thing. But on a more practical note, who do I speak to to get those instructions? (maybe Adobe/Microsoft Integration first)

**Reply by fashxp** [reply]
Installation instructions are online available. 
CI-Hub integration for example here: https://pimcore.com/docs/data-hub-ci-hub/current/Installation.html
Please read carefully, you also need to install datahub and datahub-simple-rest-API for it. 

If you need more support, please contact your sales contact person at Pimcore.

**Comment by NiklasBr** [comment]
`composer require  pimcore/enterprise-subscription-tools` solved it.

**Comment by shubhangi101092** [comment]
Hi @fashxp , can you suggest how can we validate if licence is active in local and on production server with pimcore 10.5 is there any way we can validate in admin UI or any other way

**Reply by fashxp** [reply]
<img width="482" height="143" alt="image" src="https://github.com/user-attachments/assets/fade04e8-e8a8-4770-8212-1011c1a3a198" />

If you see that, your license is not valid. If there is a green element, everything is fine. If there is nothing above the Pimcore logo, you don't have enterprise bundles installed.
